### DISCUSSION TITLE
Add custom component for relationship select item

### DISCUSSION THREAD

**Discussion by jamesvclements** [payloadcms/payload#2934] [Roadmap]
Per this discussion: https://discord.com/channels/967097582721572934/1123646638544662600/1123646638544662600

TLDR it would be really helpful to allow us to customize the DropdownItem used in `relationship` fields. Right now it defaults to the ID (or title, if set) which is generally helpful but doesn't allow you to add thumbnails. For a media or video collection sometimes the thumbnail itself is the only way to uniquely identify each item, so you need to click into the item itself to see what it's referencing.

I'm not sure whether this better belongs:
* In a **Collections** `admin.components`, so whenever that collection is used in a relationship field, it uses that component to render the items in the select.
* Or, in the **Field** `admin.components` for the relationship itself, so that each relationship can uniquely define the component that's rendered. This is probably the better option since it's more flexible.

Here's a rough mock of what this would allow, making relationship selects much easier:
![image](https://github.com/payloadcms/payload/assets/20052710/dc46fe47-d099-4dbf-977a-cc461f5b51fe)

**Comment by qbeauperin** [comment]
Great suggestion, I think this would be a very useful option to have!

My gut tells me that this should be at the collection level to avoid having to define that custom component in every relationship field (as I would imagine that the component is gonna be dependent on the collection's data more often than it will on the context of the relationship field), but maybe the solution is actually _both_? 🤔  

Define the "default" at the collection level:
```typescript
{
  slug: 'collection',
  admin: {
    components: {
      relationship: MyCustomRelationshipItem,
    }
  }
}
```
then optionally in another collection, at the field level, have the ability to override the component:
```typescript
{
  slug: 'otherCollection',
  fields: [
    {
      type: 'relationship',
      name: 'collection',
      relationTo: [
        {
          collection: 'collection',
          component: AnotherCustomRelationshipItem,
        }
      ]
    }
  ]
}
```

**Comment by clandaudev** [comment]
I have been looking for something like this as well, would love to see this implemented!

**Comment by LeonardoRick** [comment]
Yes, that would help a lot! Great idea

**Comment by Oliver-Turp** [comment]
Not sure if this is in the works but this would be a godsend for my projects!

**Comment by DanielleREG** [comment]
Bump this idea

**Comment by DiegoGonzalezCruz** [comment]
Was this ever considered?

**Comment by jmikrut** [comment]
Hey all, just chiming in here - I added this to our roadmap just now! We agree that this will be a good feature to support.

Thanks for all the interest!

**Reply by tophermurphy** [reply]
Thanks. I was just about to submit the same thing. 

You guys rock!

**Reply by robbertstevens** [reply]
@jmikrut any update on this? If not already implemented
